src/script.js

```diff
@@ -1,117 +1,50 @@
-  const li = e.target.closest('li');
-  if (li) {
-    // Toggle the "selected" class of the li element
-    li.classList.toggle('selected');
-    // Find the ellipsis and trash icons within the li element
-    const ellipsisIcon = li.querySelector('.fa-ellipsis-v');
-    const trashIcon = li.querySelector('.fa-trash');
-    if (li.classList.contains('selected')) {
-      // If the li element has the "selected" class, hide the ellipsis icon and show the trash icon
-      ellipsisIcon.style.display = 'none';
-      trashIcon.style.display = 'block';
-      trashIcon.style.cursor = 'pointer';
-    } else {
-      // If the li element does not have the "selected" class,
-      // show the ellipsis icon and hide the trash icon
-      ellipsisIcon.style.display = 'block';
-      trashIcon.style.display = 'none';
-    }
-  }
-
-  refMe.addEventListener('click', () => {
-    window.location.reload();
+function clearAllFunc() {
+  const task = tasks.filter((task) => !task.completed);
+  task.forEach((task, index) => {
+    task.index = index;
   });
-
-  function deleteTask(index) {
-    tasks.splice(index, 1); // Remove the task at the given index from the tasks array
-
-    tasks.forEach((task, index) => {
-      task.index = index; // Update the task index
-    });
-    localStorage.setItem('tasks', JSON.stringify(tasks)); // Save the tasks array to local storage
-
-    displayTasks(); // Render the updated todo list
-  }
-
-  // If the clicked element is an input element, update the completed status of the task
-  if (e.target.tagName === 'INPUT') {
-    const { index } = e.target.dataset;
-    tasks[index].completed = e.target.checked;
-    localStorage.setItem('tasks', JSON.stringify(tasks));
-    displayTasks();
-  } else if (e.target.classList.contains('fa-ellipsis-v')) {
-    // If the clicked element is the ellipsis icon, move the task up one position in the list
-    const item = e.target.parentNode;
-    const prev = item.previousElementSibling;
-    todoList.insertBefore(item, prev);
-    tasks.forEach((task, index) => {
-      task.index = index; // Update the task index
-    });
-    localStorage.setItem('tasks', JSON.stringify(tasks));
-  } else if (e.target.classList.contains('fa-trash')) {
-    // If the clicked element is the trash icon, delete the task
-    const item = e.target.parentNode;
-    const { index } = item.querySelector('input').dataset;
-    deleteTask(index);
-  }
+  localStorage.setItem('tasks', JSON.stringify(task));
+  displayTasks();
 }
 
-// Add event listeners to various elements on the page
-form.addEventListener('submit', addItem);
-todoList.addEventListener('click', toggleItem);
-enterIcon.addEventListener('click', addItem);
+form.addEventListener('submit', crud.addItem);
+todoList.addEventListener('click', crud.toggleItem);
+enterIcon.addEventListener('click', crud.addItem);
+clearAll.addEventListener('click', clearAllFunc);
+todoList.addEventListener('dblclick', crud.editTaskDescription);
```

Comment:
- [ ] [OPTIONAL] The double click can be confusing to a user, you can make it at onclick :+1: